### DISCUSSION TITLE
Authentication fails

### DISCUSSION THREAD

**Discussion by Freeeder** [microsoftgraph/msgraph-sample-copilot-plugin#12] [General]
Following the readme I ran into issues authenticating to the Copilot-Plugin

I can see the swagger , both on localhost and on the tunneled address.

When I publish the plugin to Teams, it's visible right away and I can start chatting immediately, however when it comes to the point to authenticate with the plugin it closes after a while but it requests another authentication.

Maybe seems like the redirect Uri isn't right?

Any idea on that?

**Comment by jasonjoh** [comment]
Let me start by saying it's possible that your Microsoft 365 tenant hasn't had all of the required bits rolled out yet.

> when it comes to the point to authenticate with the plugin it closes after a while but it requests another authentication.

I'm not sure I follow - do you get this?

![image](https://github.com/user-attachments/assets/823c91f6-fc37-4bbf-acc3-301594f8aca0)

Clicking **Sign in to Budget Tracker** should cause a pop-up window to do the auth and consent.

> Maybe seems like the redirect Uri isn't right?

Redirect for the plugin should be `https://teams.microsoft.com/api/platform/v1.0/oAuthRedirect`

**Reply by Freeeder** [reply]
Thanks for your quick reply.

Yes, that's excactly what happens, at least on the first time.
It asks me for username and password and the consent.

Still, after a while it asks me the to Sign in to Budget Tracker again.
This time no popup shows up, but the loading animation starts and stops after a few seconds and it asks me to sign in again.

So I'm trapped in a loop here.

The only interesting thing I found in debug mode is this:
![image](https://github.com/user-attachments/assets/b76e1c3c-4265-4ba6-a0cd-fce49e3002a6)

You can see three clicks on Sign In to Budget Tracker here.

**Reply by jasonjoh** [reply]
Does it ever successfully call the API? If so, how long is "a while" before it asks again?

Those errors look like errors calling back to the "plugin auth vault" that stores the information about your client ID and secret. If you sign into https://dev.teams.microsoft.com/tools, can you see an option for **OAuth client registration**?

**Comment by Freeeder** [comment]
No, it never actually calls the API. There is no traffic at all on the endpoint.

Interestingly, if I acquire an Authorization Code - Access Token via Postman (and adding the Postman redirect Uri), I can call the API successfully. Same settings as in the Plugin:


![image](https://github.com/user-attachments/assets/6e1705d9-e0d8-4a1d-895e-adadd0eeaa72)


**It takes between 5 and 10 seconds and then the loading animation is gone and I can click Sign In to Buget again.**


**Here is the Oauth Clien Registration: (Triple checked the input parameters. )**

![image](https://github.com/user-attachments/assets/aa8d4ff2-5866-4ed0-a043-17176e788004)



I Can find the ID in the browser:

![image](https://github.com/user-attachments/assets/80cd9816-f236-4f7a-9f7a-d03608f3a2a2)


and in the App Manifest dev JSON:

![image](https://github.com/user-attachments/assets/526c98d5-eb3b-4131-adf1-1ef568ddc86e)

**Reply by jasonjoh** [reply]
Yeah, I would expect it to work with a token in Postman or similar. The issue it appears is with the auth calls back to the plugin vault. For that error you showed, is it returning a 40x error, and were there any response headers you can share? If you have a value for the `Traceresponse` header I can see if I can get any information, but I'm not in support so I have limited ability here.

Also just to verify, on your plugin app registration, you have the `access_as_user` scope for your API set in **API Permissions**?

![image](https://github.com/user-attachments/assets/d74a604c-3dd9-4d73-9129-4e1d2ec76e74)

**Reply by Freeeder** [reply]
**Actually it's returning a 200 but with an error in the message:**
TraceResponse: 00-154a910e1a705f5f89aab40046847085-1541eef0be46d856-00

![image](https://github.com/user-attachments/assets/8fb22661-993b-4b04-8bff-66f69bd2d4cd)

**Response:**


![image](https://github.com/user-attachments/assets/5a295a7f-dc8c-4db3-99be-8b759f434c5b)

**This is the permission config of the plugin app registration:**
![image](https://github.com/user-attachments/assets/9412ed04-97c2-4c73-8c40-6b8fad4b37b6)
